OpenRA.Mods.Common/Activities/LayMines.cs

```diff
@@ -136,27 +147,42 @@ static bool CanLayMine(Actor self, CPos p)
 			return self.World.ActorMap.GetActorsAt(p).All(a => a == self);
 		}
 
-		void LayMine(Actor self)
+		bool StartLayingMine(Actor self)
 		{
 			if (ammoPools != null)
 			{
 				var pool = ammoPools.FirstOrDefault(x => x.Info.Name == minelayer.Info.AmmoPoolName);
 				if (pool == null)
-					return;
+					return false;
 
 				pool.TakeAmmo(self, minelayer.Info.AmmoUsage);
 			}
 
 			foreach (var t in self.TraitsImplementing<INotifyMineLaying>())
 				t.MineLaying(self, self.Location);
 
+			if (minelayer.Info.LayMineTime == 0)
+			{
+				FinishLayingMine(self);
+				return true;
+			}
+
+			return false;
```

Comment:
It's weird here to return `false` even though it did start laying a mine. I suggest always returning true and moving the `minelayer.Info.LayMineTime == 0` check up a call. (See next comment.)